my-sms/spec/controllers/students_controller_spec.rb

```diff
@@ -0,0 +1,136 @@
+
+require 'rails_helper'
+
+# This spec was generated by rspec-rails when you ran the scaffold generator.
+# It demonstrates how one might use RSpec to specify the controller code that
+# was generated by Rails when you ran the scaffold generator.
+#
+# It assumes that the implementation code is generated by the rails scaffold
+# generator.  If you are using any extension libraries to generate different
+# controller code, this generated spec may or may not pass.
+#
+# It only uses APIs available in rails and/or rspec-rails.  There are a number
+# of tools you can use to make these specs even more expressive, but we're
+# sticking to rails and rspec-rails APIs to keep things simple and stable.
+#
+# Compared to earlier versions of this generator, there is very limited use of
+# stubs and message expectations in this spec.  Stubs are only used when there
+# is no simpler way to get a handle on the object needed for the example.
+# Message expectations are only used when there is no simpler way to specify
+# that an instance is receiving a specific message.
+#
+# Also compared to earlier versions of this generator, there are no longer any
+# expectations of assigns and templates rendered. These features have been
+# removed from Rails core in Rails 5, but can be added back in via the
+# `rails-controller-testing` gem.
+
+RSpec.describe StudentsController, type: :controller do
+  # This should return the minimal set of attributes required to create a valid
+  # Student. As you add validations to Student, be sure to
+  # adjust the attributes here as well.
+
+  # This should return the minimal set of values that should be in the session
+  # in order to pass any filters (e.g. authentication) defined in
+  # StudentsController. Be sure to keep this updated too.
+  let(:valid_session) { {} }
+
+  describe 'GET #index' do
+    it 'returns a success response' do
+      get :index, {}, valid_session
+      expect(response).to be_success
+    end
+  end
+
+  describe 'GET #show' do
+    it 'returns a success response' do
+      student = FactoryBot.create(:student)
+      get :show, { id: student.to_param }, valid_session
+      expect(response).to be_success
+    end
+  end
+
+  describe 'GET #new' do
+    it 'returns a success response' do
+      get :new, {}, valid_session
+      expect(response).to be_success
+    end
+  end
+
+  describe 'GET #edit' do
+    it 'returns a success response' do
+      student = FactoryBot.create(:student)
+      get :edit, { id: student.to_param }, valid_session
+      expect(response).to be_success
+    end
+  end
+
+  describe 'POST #create' do
+    context 'with valid params' do
+      it 'creates a new Student' do
+        expect do
+          post :create, { student: FactoryBot.attributes_for(:student) }, valid_session
+        end.to change(Student, :count).by(1)
+      end
+
+      it 'redirects to the created student' do
+        post :create, { student: FactoryBot.attributes_for(:student) }, valid_session
```

Comment:
Here the `subject {...}` pattern would make your tests more DRY (not repeating code unnecessarily)